frontend-ssr/pages/api/user/register.js

```diff
@@ -0,0 +1,56 @@
+//Imports 
+
+const User = require("../../../models/User.schema");
+import dbConnect from "../../../utils/dbConnect";
+const bcrypt = require("bcryptjs")
+dbConnect();
+const {registerValidation} = require("../../../utils/validation/userValidation") 
+// import UserRequests from "../../../utils/user.requests";
+
+export default async (req, res) => {
+    let { method, body } = { ...req };
+
+    switch (method) {
+        case "POST":
+        //Desctruct and create vars    
+            // console.log(method, body);
+            let {name,username,avatar,email,password}={...body}
+            let {error} = registerValidation(body);
+            if (error)return res.status(400).json({"message":error})
+            let errors=[]
+            // Encrypt the password 
+            const salt= await bcrypt.genSalt(10); 
+            const hash_password= await bcrypt.hash(password,salt)
+            //Create the user for the database 
+            let created_user=new User({
+                name:name,
+                username:username,
+                avatar:avatar,
+                email:email,
+                password:hash_password
+            });
+            //Check if the email and username already exists
+            const emailExists=await User.findOne({email:body.email});
+            if(emailExists){ errors.push({email:"Email already exists"});}
```

Comment:
In the `login` route you extracted the error messages in their separate variables. I think that approach is better so let's do this here as well. 

Also, same feedback regarding `await try-catch'es`. Let's handle them

Reply:
Surrounded with try-catch and moved error to separate variables to handle errors

Reply:
Thanks. Just a reminder to insert the CAPS_LOCK feedback here is well.